gcovr/gcov.py

```diff
@@ -604,6 +605,73 @@ def check_unrecognized_lines(self, ignore_parse_errors):
         sys.exit(1)
 
 
+class gcovr_executer:
+    gcov_cmd = None
+    gcov_options = None
+
+    def __init__(self, options):
+        # If the first element of cmd - the executable name - has embedded spaces
+        # (other than within quotes), it probably includes extra arguments.
+        if gcovr_executer.gcov_cmd is None:
+            gcovr_executer.gcov_cmd = shlex.split(options.gcov_cmd)
+            gcovr_executer.gcov_options = [
```

Comment:
Do these really have to be class variables (globals)? The code looks like assigning to `self` fields would work.

Reply:
First I used global variables to safe time because clang doesn't understand the `--demangle-names`.
Maybe it's better to revert this part and spend the time to build the commandline several times. I think the safed time is less in comparsion to the time gcov needs.

Reply:
Rebuilding the command is essentially free, especially compared to I/O like launching a gcov process.

Using a class like this is probably a good idea, but only if it simplifies things. I find the data flow from `options` to the *global* `gcov_cmd` and `gcov_options` variables very confusing. Potentially, it might be possible to change the class variables to instance variables and to push the `executor` construction further outward, perhaps to the main(). This would have the same caching effect, clear data flows, but would involve extra function parameters to pass the executor around.

Reply:
I think this is a bigger change and something for the future. E.G. to define a interface for different input formats. I'll revert the change here.